src/pgrn-variables.c

```diff
@@ -103,19 +104,14 @@ PGrnLogTypeAssign(int new_value, void *extra)
 		grn_logger_set(ctx, &PGrnPostgreSQLLogger);
 		break;
 	default:
-#if !GRN_VERSION_OR_LATER(14, 0, 8)
 		/*
 		 * grn_logger_set() is not run to set the default value.
 		 * (= Do not run grn_logger_set() before PGrnGroongaInitialized.)
 		 * Because it crashes on Windows under Groonga 14.0.8.
 		 */
-		if (PGrnGroongaInitialized)
-		{
-#endif
-			grn_logger_set(ctx, NULL);
-#if !GRN_VERSION_OR_LATER(14, 0, 8)
-		}
-#endif
+		if (!PGrnGroongaInitialized && !PGrnGroongaVersionOrLater(14, 0, 8))
+			break;
+		grn_logger_set(ctx, NULL);
```

Comment:
こっちの方がNOTが減ってわかりやすくないかな。

```suggestion
		if (PGrnGroongaVersionOrLater(14, 0, 8) || PGrnGroongaInitialized)
			grn_logger_set(ctx, NULL);
```

Reply:
何もしないことのアピールのためにNOTの条件にしてましたが、NOTしないほうがだいぶわかりやすいですね。
NOTの条件が正しいか考えるのにだいぶ脳みそを使いました…。